RELEASING.md

```diff
@@ -24,95 +24,211 @@ tools must be installed.
+
+### Versions
+
+This command will create a commit that updates the version from `x.y.z-SNAPSHOT` to `x.y.z`.
+The current `pom.xml` files will be backed up with the name `pom.xml.releaseBackup` and left in staging.
+
+```Shell
+mvn clean release:prepare
+```
+
+### Pull Request
+
+Open up pull requests for the documentation and version updates.
+After the updates are approved and merged, continue on with the release process.
+
+
+### GPG Signing Key
+
+The JanusGraph artifacts are signed with a GPG signature.
+If you don't already have a GPG signing key included with the `KEYS` file, you will need to create one and update the `KEYS` file.
+The `KEYS` file is posted on the releases page.
+
+Build Release Artifacts
+-----------------------
+
+* Pull down the latest, merged code from GitHub.
+* Stash any uncommitted changes.
+* Delete untracked files and directories.
+* Deploy it.
+
+```Shell
+export JG_VER="janusgraph-0.2.3"
+git fetch
+git pull
+git stash save
+git clean -fd
+cd janusgraph-dist
+mvn clean javadoc:jar deploy -Pjanusgraph-release -DskipTests=true
+cp janusgraph-dist/janusgraph-dist-hadoop-2/target/${JG_VER}-hadoop2.zip* ~/jg-staging/
+cd janusgraph-doc/target/docs/
+cd target/docs; mv chunk ${JG_VER}-hadoop2-doc
+zip -r ${JG_VER}-hadoop2-doc.zip ${JG_VER}-hadoop2-doc
+gpg --armor --detach-sign ${JG_VER}-hadoop2-doc.zip
+cp ${JG_VER}-hadoop2-doc.zip* ~/jg-staging/
+cd ~/jg-staging
+gpg --verify ${JG_VER}-hadoop2.zip.asc ${JG_VER}-hadoop2.zip
+```
+
+If it fails due to Inappropriate ioctl for device error, run:
+```Shell
+export GPG_TTY=$(tty)
 ```
 
-### Checking Artifacts & Archives
+### Close the staging repository
 
-This is a good time to inspect the archives just uploaded to S3. Directory
-listing is normally disabled on that directory, so you may need to login to the
-S3 console to list the contents and check that nothing's out of place.
+Log into https://oss.sonatype.org/#welcome and select Staging Repositories under Build Promotion.
+If you recently uploaded you can easily find your staged release by doing a descending sort.
+Verify that the contents look complete and then check the release before clicking close.
 
-If S3 looks good, then inspect and close the staging repository that
-the deploy phase automatically created on https://oss.sonatype.org/.
+This step will run verification on all the artifacts.
+In particular, this will verify that all of the artifacts have a valid GPG signature.
+If this step fails, there are changes that must be made before starting the vote.
 
-### Pushing the Release
+## Finalize the Release
 
 *This is the point of no return.* After releasing artifacts to Maven
 Central and pushing history to the public Github repo, the release
 can't be canceled.
 
-Release the Sonatype OSS repository that you inspected and closed
-earlier.  It will appear on Maven Central in an hour or two.
+### Release the staging repository
 
-Finally, push your local changes to Github:
```

Comment:
Where are the changes now pushed to GitHub? (I just can't find this step any more in the updated document.)

Reply:
This is back from the Titan repo. This might be referring to uploading the artifacts for the release, but I didn't see anything that directly aligned in Jason's notes.